src/payment/strategies/adyenv3/adyenv3.ts

```diff
@@ -0,0 +1,827 @@
+
+    /*
+     * The Client Key of your Adyen account.
+     */
+    clientKey?: string;
+
+    /*
+     * Supported from Components version 3.0.0 and later. The full paymentMethods response,
+     * returned in step 1. We recommend that you pass this on the AdyenCheckout instance.
+     * Otherwise, you need to pass the specific payment method details separately for each
+     * Component.
+     */
+    paymentMethodsResponse?: PaymentMethodsResponse;
+
+    showPayButton?: boolean;
+
+    /*
+     * Specify the function that you created, for example, handleOnChange. If you wish
+     * to override this function, you can also define an onChange event on the Component
+     * level.
+     */
+    onChange?(state: CardState, component?: AdyenComponent): void;
+
+    onAdditionalDetails?(state: CardState, component?: AdyenComponent): void;
+}
+
+export interface AdyenPlaceholderData {
+    holderName?: string;
+    billingAddress?: {
+        street: string;
+        houseNumberOrName: string;
+        postalCode: string;
+        city: string;
+        stateOrProvince: string;
+        country: string;
+    };
+}
+
+export interface AdyenV3CreditCardComponentOptions extends AdyenBaseCardComponentOptions, AdyenComponentEvents {
+    /**
+     * Set an object containing the details array for type: scheme from
+     * the /paymentMethods response.
+     */
+    details?: InputDetail[];
+
+    /**
+     * Set to true to show the checkbox to save card details for the next payment.
+     */
+    enableStoreDetails?: boolean;
+
+    /**
+     * Set to true to request the name of the card holder.
+     */
+    hasHolderName?: boolean;
+
+    /**
+     * Set to true to require the card holder name.
+     */
+    holderNameRequired?: boolean;
+
+    /**
+     * Information to prefill fields.
+     */
+    data?: AdyenPlaceholderData;
+
+    /**
+     * Defaults to ['mc','visa','amex']. Configure supported card types to
+     * facilitate brand recognition used in the Secured Fields onBrand callback.
+     * See list of available card types. If a shopper enters a card type not
+     * specified in the GroupTypes configuration, the onBrand callback will not be invoked.
+     */
+    groupTypes?: string[];
+
+    /**
+     * Specify the sample values you want to appear for card detail input fields.
+     */
+    placeholders?: CreditCardPlaceHolder | SepaPlaceHolder;
+}
+
+export interface AdyenCustomCardComponentOptions extends AdyenBaseCardComponentOptions, AdyenComponentEvents {
+    /**
+     * Specify aria attributes for the input fields for web accessibility.
+     */
+    ariaLabels?: CustomCardAriaLabels;
+
+    /**
+     * Automatically shift the focus from date field to the CVC field.
+     */
+    autofocus?: boolean;
+}
+
+export interface AdyenError {
+    errorCode: string;
+    message: string;
+}
+
+export interface AdyenHostWindow extends Window {
+    AdyenCheckout?: new(configuration: AdyenConfiguration) => AdyenClient;
```

Comment:
This is marked as optional, is that supposed to be?

Reply:
Can you please take a look at this page and see if it helps https://github.com/Adyen/adyen-web/issues/1251